File: evals/refs/pow2.zig
const std = @import("std");

pub fn main(init: std.process.Init) !void {
    var in_buf: [4096]u8 = undefined;
    var fr = std.Io.File.stdin().reader(init.io, &in_buf);
    const r = &fr.interface;

    var out_buf: [4096]u8 = undefined;
    var fw = std.Io.File.stdout().writer(init.io, &out_buf);
    const w = &fw.interface;

    while (try r.takeDelimiter('\n')) |line| {
        const n = std.fmt.parseInt(i64, line, 10) catch {
            try w.writeAll("false\n");
            continue;
        };
        const is_pow = n > 0 and (n & (n - 1)) == 0;
        try w.writeAll(if (is_pow) "true\n" else "false\n");
    }
    try w.flush();
}

File: evals/refs/pow2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Buffer(make([]byte, 0, 1024*1024), 1024*1024)
	w := bufio.NewWriter(os.Stdout)
	defer w.Flush()
	for sc.Scan() {
		line := strings.TrimSpace(sc.Text())
		if line == "" {
			continue
		}
		n, err := strconv.ParseInt(line, 10, 64)
		if err != nil {
			fmt.Fprintln(w, "false")
			continue
		}
		if n > 0 && (n&(n-1)) == 0 {
			fmt.Fprintln(w, "true")
		} else {
			fmt.Fprintln(w, "false")
		}
	}
}
